happytransformer/happy_generation.py

```diff
@@ -8,6 +8,74 @@
 from happytransformer.toc.trainer import TOCTrainer
 from happytransformer.adaptors import get_adaptor
 
+gen_greedy_settings = {
```

Comment:
constants should be `ALL_UPPERCASE`

Reply:
Good point. 

But, I imagine users will want to copy the predefined settings and then make changes. For example, increase "top_k" when using "gen_top_k_sampling_settings."

Should we still rename them to ALL_UPPERCASE?

Reply:
Users can still copy the settings if they are constants. Are you talking about people mutating the built in settings defaults?

Reply:
Yes they can copy them. No, I'm more referring to giving the false impression that that they can not be changed. We want users to import the predefined settings and then make changes as they wish.

Reply:
The predefined settings being constant doesn't imply that only predefined settings are allowed. If your concern is with users not knowing that custom greedy settings are allowed, then we could call these `DEFAULT_GREEDY_SETTINGS`

Reply:
I like your idea of using "default" rather than "gen." 

I still think it's better to go with all lowercase, because we want to encourage users to modify the imported settings. Also, although we don't modify the provided settings directly, we use it to create a new dictionary that contains settings during preprocessing. This way, we add the default values for any neglected keys. We may add more preprocessing in the future as well that may modify the dictionary dictionary directly.